src/main.ts

```diff
@@ -1360,7 +1367,10 @@ function generateNestjsServiceClient(
 
     // Use metadata as last argument for interface only configuration
     if (options.addGrpcMetadata) {
-      requestFn = requestFn.addParameter('metadata?', 'Metadata@grpc');
+      requestFn = requestFn.addParameter(options.addRestParameter ? 'metadata' : 'metadata?', 'Metadata@grpc');
+    }
+    if (options.addRestParameter) {
+      requestFn = requestFn.addParameter('...rest', 'any');
```

Comment:
I was thinking it'd be nice if could directly be generated as `user: User` here, but I get that these annotations are very specifically something that user adds one-by-one-sy to each call site as needed. So the `any` makes sense.

Reply:
@stephenh I don't think we wanna generate this directly, cause it needs to be dynamic that's why the `...rest: any` comes in handy.